service/src/main/java/org/jboss/sbomer/service/feature/sbom/service/AdvisoryService.java

```diff
@@ -167,6 +243,61 @@ private Collection<SbomGenerationRequest> handleStandardQEAdvisory(RequestEvent
         }
     }
 
+    @Transactional
+    protected Collection<SbomGenerationRequest> processRequestConfigsWithinNotes(
+            RequestEvent requestEvent,
+            List<RequestConfig> requestConfigsWithinNotes) {
+
+        List<SbomGenerationRequest> generations = new ArrayList<SbomGenerationRequest>();
+        for (RequestConfig config : requestConfigsWithinNotes) {
+            if (config instanceof PncBuildRequestConfig) {
+                log.info("New PNC build request received");
+
+                generations.add(sbomService.generateFromBuild(requestEvent, config, null));
+            } else if (config instanceof PncAnalysisRequestConfig analysisConfig) {
+                log.info("New PNC analysis request received");
+
+                generations.add(
+                        sbomService.generateNewOperation(
+                                requestEvent,
+                                DeliverableAnalysisConfig.builder()
+                                        .withDeliverableUrls(analysisConfig.getUrls())
+                                        .withMilestoneId(analysisConfig.getMilestoneId())
+                                        .build()));
+            } else if (config instanceof PncOperationRequestConfig operationConfig) {
+                log.info("New PNC operation request received");
+
+                generations.add(
+                        sbomService.generateFromOperation(
+                                requestEvent,
+                                OperationConfig.builder().withOperationId(operationConfig.getOperationId()).build()));
+            }
+        }
+
+        return generations;
+    }
+
+    /**
+     * Helper method to create RequestConfig based on the deliverable type.
+     */
+    private RequestConfig createRequestConfig(String type, JsonNode deliverableEntry) {
```

Comment:
It's ok for now, but I would like to move this https://github.com/project-ncl/sbomer/blob/e560e9462a0768b9811efcdc785c1514af0bcd27/core/src/main/java/org/jboss/sbomer/core/features/sbom/config/Config.java#L121-L221 to a utility class and use instead of what we have here.

It provides much more detailed information in case of failures.